Content.Client/Anomaly/AnomalyScannerSystem.cs

```diff
@@ -0,0 +1,94 @@
+using Content.Shared.Anomaly;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace Content.Client.Anomaly;
+
+/// <inheritdoc cref="SharedAnomalyScannerSystem"/>
+public sealed class AnomalyScannerSystem : SharedAnomalyScannerSystem
+{
+    [Dependency] private readonly IClyde _clyde = default!;
+    [Dependency] private readonly SpriteSystem _sprite = default!;
+
+    // Just an array to initialize the pixels of a new OwnedTexture
+    private static readonly Rgba32[] EmptyTexture = new Rgba32[32*32];
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<AnomalyScannerScreenComponent, ComponentInit>(OnComponentInit);
+        SubscribeLocalEvent<AnomalyScannerScreenComponent, ComponentStartup>(OnComponentStartup);
+        SubscribeLocalEvent<AnomalyScannerScreenComponent, AppearanceChangeEvent>(OnScannerAppearanceChanged);
+    }
+
+    private void OnComponentInit(Entity<AnomalyScannerScreenComponent> ent, ref ComponentInit args)
+    {
+        // Allocate the OwnedTexture
+        ent.Comp.ScreenTexture ??= _clyde.CreateBlankTexture<Rgba32>((32, 32));
+        // Initialize the texture
+        ent.Comp.ScreenTexture.SetSubImage((0, 0), (32, 32), new ReadOnlySpan<Rgba32>(EmptyTexture));
+
+        // Initialize bar drawing buffer
+        ent.Comp.BarBuf ??= new Rgba32[ent.Comp.Size.X * ent.Comp.Size.Y];
+    }
+
+    private void OnComponentStartup(Entity<AnomalyScannerScreenComponent> ent, ref ComponentStartup args)
+    {
+        if (!TryComp<SpriteComponent>(ent, out var sprite))
+            return;
+
+        _sprite.LayerSetTexture((ent, sprite), AnomalyScannerVisualLayers.Screen, ent.Comp.ScreenTexture);
+    }
+
+    private void OnScannerAppearanceChanged(Entity<AnomalyScannerScreenComponent> ent, ref AppearanceChangeEvent args)
+    {
+        args.AppearanceData.TryGetValue(AnomalyScannerVisuals.AnomalySeverity, out var severityObj);
+        if (severityObj is not float severity)
+            severity = 0;
+
+        if (args.Sprite is null)
+            return;
+
+        ent.Comp.ScreenTexture ??= _clyde.CreateBlankTexture<Rgba32>((32, 32));
+
+        if(ent.Comp.BarBuf?.Length != ent.Comp.Size.X * ent.Comp.Size.Y)
+            ent.Comp.BarBuf = new Rgba32[ent.Comp.Size.X * ent.Comp.Size.Y];
+
+        // Get the bar length
+        var barLength = (int)(severity * ent.Comp.Size.X);
+
+        // Calculate the bar color
+        // Hue "angle" of two colors to interpolate between depending on severity
+        const float greenHue = 110f / 360f;
+        const float redHue = 0f;
+        // Just a lerp from greenHue at severity = 0.5 to redHue at 1.0
+        var hue = Math.Clamp(2*greenHue * (1 - severity), redHue, greenHue);
+        var color = new Rgba32(Color.FromHsv((hue, 1, 1, 1)).RGBA);
```

Comment:
all of the constants here should be de-magic-numbered, ideally they'd be datafields on the component

you should also interpolate brightness as well as hueshifting alone isn't great for colourblind legibility, with darker being less severe and brighter when it's more severe

Reply:
The color is secondary to the length of the bar, even if it was colorless, the length of the bar would convey the full severity information. I'll fix the constants